data_steward/cdr_cleaner/cleaning_rules/drop_extreme_measurements.py

```diff
@@ -0,0 +1,115 @@
+"""
+Background
+
+Background: For physical measurements collected by the Program, warnings or limits exist for height, weight, and BMI;
+however, there are still some outliers in the dataset from before the addition of these limits (or in cases where limits
+are more relaxed). These outliers need to be dropped to improve data quality.
+
+Scope: Create the following cleaning rule (all Weights are in KILOGRAMS all Heights are in CENTIMETERS): In the
+measurements table, extreme values of height, weight, and BMI should be dropped along with related values for the
+participant.
+
+This is expected to drop a very small number of rows (less than 300 total) based on values in the current CDR.
+
+"""
+
+# Project imports
+import constants.cdr_cleaner.clean_cdr as cdr_consts
+
+DELETE_HEIGHT_ROWS_QUERY = """
+--drop all height recrods out of bounds
+DELETE
+FROM `{project_id}.{dataset_id}.measurement`
+WHERE measurement_source_concept_id = 903133
+AND value_as_number NOT BETWEEN 90 AND 228
+UNION ALL
+--drop BMI row associated with PID
+DELETE
+FROM `{project_id}.{dataset_id}.measurement`
+WHERE person_id IN (SELECT person_id 
```

Comment:
I saw that the ticket requests related rows to be dropped (for bad BMI, it says drop the height and weight rows for that pid). However, it does not clarify a few things:
- If one BMI record is bad, should all BMI records for that pid be dropped?
- If one BMI record is bad, should all height and weight rows for that pid be dropped or just one each? Also how to find the height and weight rows used to calculate the BMI row? Are they all related via timestamps?

Reply:
After receiving clarification, it looks like the only link between the bad records in a particular measure (say height) and the corresponding records in the other measures (BMI/weight) is the timestamp.
I believe there are many ways to tackle this. One way I can think of would be to have a subquery which only retrieves the person_id, timestamp and the extreme value for a particular measure (say height), and the main query choosing measurement_ids for the same person_ids and same timestamps but for all measures (height/BMI/weight), and the delete statement deleting these measurement_ids. May not be the most practical way to do it so feel free to solve it your way.